Kudu.Core/SourceControl/Git/LibGit2SharpRepository.cs

```diff
@@ -148,8 +148,13 @@ public void AddFile(string path)
             }
         }
 
-        public bool Commit(string message, string authorName, string emailAddress)
-        {
+        public bool Commit(string message, string authorName, string emailAddress, string commitId = null)
```

Comment:
If NullRepository constructor can take the commitId param, all Commit() functions can retain the old signature.. Can you check if it is possible?

Reply:
Seems to be. Will update.

Reply:
Would it not be a better idea to add it to this method? Since if and when we want to add support for any other deployment kind they would all follow the same contract that way